src/controllers/envelopes-controller.ts

```diff
@@ -20,6 +19,18 @@ function findEnvelopeIndex(id: string): number {
   });
 }
 
+// Middleware that sets request object property
+export const setEnvelopeIndex: RequestParamHandler = (req, res, next, id) => {
+  const foundIndex = envelopes.findIndex((item) => item.id === id);
+  if (foundIndex === -1) {
+    return res
+      .status(404)
+      .json({ message: `Couldn't find Envelope id: ${id}` });
+  }
+  req.envelopeIndex = foundIndex;
```

Comment:
We should probably be explicit that this is a value we validated, e.g. `req.validatedEvelopeIndex`. During my first read I missed this and later thought this was a user-provided value.

Reply:
Absolutely, makes sense! :)